spec/formatting.md

```diff
@@ -174,7 +177,7 @@ the following steps are taken:
 
 ### Fallback Resolution
 
-The resolution of an _expression_ may fail in the following cases:
+The resolution of an _expression_ MAY fail in the following cases:
```

Comment:
Do we really mean _MAY_ here? This isn't normative permission, really. In this case, if one of the following things happens, the expression almost certainly fails to resolve. (To be fair, it might not fail, if the failure were something that was recoverable or irrelevant, e.g. `{$foo :bar option=$unknown}` where `option` is an option that `:bar` doesn't care about).

I think this section is slightly upside down. I'd probably change it by bringing line 186 forward and modifying it to actually define the term in question, kind of like so:

> A **_fallback value_** is the value emitted when an _expression_ cannot be resolved.
>
> Some of the reasons that an _expression_ might fail to be resolved include:
> * A _variable_ _operand_ fails to resolve.
> * A _function_'s _annotation_ fails to resolve.
> * The _expression_ contains or consists of a _reserved_ _annotation_.
>
> The _fallback value_ depends on the content of the _expression_:

(Note that the last line of my suggestion replaces an occurrence of the word "shape" :wink in the current text)

Reply:
The list here is intended to be exhaustive, so I would say that MAY is appropriate. As in, we do not allow for failure in literal resolution, and failure to resolve a variable option value does not directly cause an annotation failure. The MAY is allowing for expression resolution failure in the listed cases, but not otherwise.

Reply:
The word "may" followed by a list of cases will not be understood by native speakers as being exhaustive (aka complete). Moreover, MAY is not the right term to use in such a case. 

"MAY   This word, or the adjective "OPTIONAL", mean that an item is
   truly optional.  One vendor may choose to include the item because a
   particular marketplace requires it or because the vendor feels that
   it enhances the product while another vendor may omit the same item." 

What you are conveying is

* when there is a _variable_ _operand_ that fails to resolve, the implementation might or might not resolve an expression
* when there is a _function_ _annotation_ that fails to resolve, the implementation might or might not resolve an expression
* in other circumstances as well, the implementation might or might not resolve an expression.

It depends on what you are trying to say. If you want to say that the resolution **only** fails in these two situations, and it **does** always fail in either of these two situations, you'd write:

The resolution of an _expression _ MUST_ fail in all and only the following cases: 

But as Addison says, it is probably better to just reorganize the text.

...

Reply:
@eemeli 

> The list here is intended to be exhaustive, so I would say that MAY is appropriate. As in, we do not allow for failure in literal resolution, and failure to resolve a variable option value does not directly cause an annotation failure. The MAY is allowing for expression resolution failure in the listed cases, but not otherwise.

If the list is exhaustive, then the normative term to use is `MUST`, as in:

> The resolution of an _expression_ MUST fail if any of the following conditions are met:

But I would still recommend my reformulation. To make the list exhaustive, reword my suggestion:

```
Some of the reasons that an _expression_ might fail to be resolved include:
```

to say:

```
An `expression` fails to resolve when:
```

Reply:
Point taken; I was wrong. Applied changes following @aphillips suggestions.